packages/platforms/delta/arm64/tn48m-dn/src/arm64_delta_tn48m_dn/module/src/psui.c

```diff
@@ -368,6 +368,8 @@ onlp_psui_info_get(onlp_oid_t id, onlp_psu_info_t* info)
         // TODO : auto detect AC / DC type
         // Just do a guess
         info->caps |= _psu_vin_type_guess(info->mvin);
+    } else {
+        info->caps |= ONLP_PSU_CAPS_AC;
```

Comment:
That’s much better than in the previous version. Thank you!

I still am not convinced, that mixing configuration and code is the best design, and would add the information to the struct.

In `platform_lib.h`:

```
typedef enum plat_psu_type {
    PLAT_PSU_TYPE_AC = 0,
    PLAT_PSU_TYPE_DC,
    PLAT_PSU_TYPE_MAX
} plat_psu_type_t;
```

Then in the PSU configuration above:

    .type = PLAT_PSU_TYPE_AC;

And then here:

    if (psu->type == PLAT_PSU_TYPE_AC)
        info->caps |= ONLP_PSU_CAPS_AC;

Reply:
@paulmenzel, you are right, I updated the codes again, and hope it make more sense now.